examples/wagemos-cosmoskit-nextjs/src/app/_providers/cosmos-kit.tsx

```diff
@@ -10,20 +10,65 @@ import { ReactNode } from 'react'
 export function CosmosKitProvider(props: { children: ReactNode }) {
   const signerOptions = {
     signingCosmwasm: (chain: Chain) => {
+      let gasTokenName = ''
       switch (chain.chain_name) {
+        case 'localjuno':
+        case 'junotestnet':
+          return { gasPrice: GasPrice.fromString('0.1ujunox') }
+        case 'juno':
+          return { gasPrice: GasPrice.fromString('0.1ujuno') }
+        case 'osmosis':
+        case 'osmosistestnet':
+        case 'osmosistestnet5':
+          return { gasPrice: GasPrice.fromString('0.035uosmo') }
         case 'neutron':
-          return {
-            gasPrice: GasPrice.fromString('0.01untrn'),
-          }
+          gasTokenName = 'untrn'
+          return { gasPrice: GasPrice.fromString(`.5${gasTokenName}`) }
+        case 'neutrontestnet':
+          gasTokenName = 'untrn'
+          break
+        case 'archwaytestnet':
+          gasTokenName = 'aconst'
+          return { gasPrice: GasPrice.fromString(`${1e12}${gasTokenName}`) }
+        case 'archway':
+          gasTokenName = 'aarch'
+          return { gasPrice: GasPrice.fromString(`${1e12}${gasTokenName}`) }
       }
     },
   }
   return (
     <ChainProvider
-      chains={chains}
+      chains={chains.map((chain) => {
+        const hasRpcs = !!chain.apis?.rpc?.length
+        return {
+          ...chain,
+          apis: {
+            ...chain.apis,
+            rpc: hasRpcs
+              ? [
+                  chain.apis!.rpc!.find((rpc) =>
+                    rpc.address.includes('polkachu'),
+                  ) ?? chain.apis!.rpc![0],
+                ]
+              : [],
+          },
+        }
```

Comment:
```suggestion
        return {
          ...chain,
          apis: {
            ...chain.apis,
            rpc: !!chain.apis?.rpc?.length
              ? [
                  chain.apis.rpc.find((rpc) =>
                    rpc.address.includes('polkachu'),
                  ) ?? chain.apis.rpc[0],
                ]
              : [],
          },
        }
```
i don't think type assertions are needed here, as if `hasRpcs` is `true`, `chain.apis.rpc` should be defined by definition. as the `hasRpcs` is not being reused, i would suggest moving it here.